src/replication.c

```diff
@@ -831,32 +1024,78 @@ ask_replica_status_all(spec_t *spec)
 	MTX_UNLOCK(&spec->rq_mtx);
 }
 
-/*
- * This function processes the status response received from replica
- * and update spec's healthy/degrade count
- */
+static void
+update_rebuild_info(spec_t *spec, mgmt_ack_t *ack_data)
+{
+
+	uint64_t data_len = 0;
+
+	MTX_LOCK(&spec->rq_mtx);
```

Comment:
all the functions related to rebuild in this PR seems to be done by mgmt_thread itself. So, this lock may not be required.

Reply:
I got your point sir..